crates/uv/tests/it/edit.rs

```diff
@@ -10194,3 +10194,198 @@ fn add_unsupported_git_scheme() {
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     "###);
 }
+
+/// In auth-policy "always", the normal authentication flow should still work.
+#[test]
+fn add_auth_policy_always_with_credentials() -> Result<()> {
+    let context = TestContext::new("3.12");
+
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(indoc! { r#"
+        [project]
+        name = "foo"
+        version = "1.0.0"
+        requires-python = ">=3.11, <4"
+        dependencies = []
+
+        [[tool.uv.index]]
+        name = "my-index"
+        url = "https://pypi-proxy.fly.dev/basic-auth/simple"
+        auth-policy = "always"
```

Comment:
What motivates this user-facing name? I think it's rare for us to have "policy" be user-facing, even if we're using it internally.

My only concern for something short like `auth = "always"` is that we may want to use that key to select a specific authentication method in the future, e.g., `auth = "keyring"` /  `auth = "netrc"` / `auth = "env"`? I guess that'd be equivalent to "always" with a specific source? Do we need separate options for those?

Reply:
We've decided to go with `authentication`.

Reply:
vs `authenticate`? (which is what I sent in Discord)